ipn/ipnlocal/local.go

```diff
@@ -1327,6 +1346,9 @@ func setExitNodeID(prefs *ipn.Prefs, nm *netmap.NetworkMap) (prefsChanged bool)
 			// reference it directly for next time.
 			prefs.ExitNodeID = peer.StableID()
 			prefs.ExitNodeIP = netip.Addr{}
+			if oldExitNodeID == prefs.ExitNodeID {
+				return false
+			}
 			return true
```

Comment:
This can be simplified down to

```suggestion
			return oldExitNodeID != prefs.ExitNodeID
```